rgdps/utilities/gmdps_converter.py

```diff
@@ -0,0 +1,496 @@
+            length=LevelLength(level["levelLength"]),
+            official_song_id=official_song_id,
+            custom_song_id=custom_song_id,
+            copy_password=level["password"],
+            original_id=level["original"],
+            two_player=level["twoPlayer"] > 0,
+            object_count=abs(level["objects"]),
+            coins=level["coins"],
+            requested_stars=level["requestedStars"],
+            render_str=level["extraString"],
+            difficulty=LevelDifficulty(level["starDifficulty"]),
+            downloads=level["downloads"],
+            likes=level["likes"],
+            stars=level["starStars"],
+            # NOTE: Only upload ts is varchar.
+            upload_ts=from_unix_ts(int(level["uploadDate"])),
+            update_ts=from_unix_ts(level["updateDate"]),
+            coins_verified=level["starCoins"] > 0,
+            feature_order=level["starFeatured"],
+            deleted=level["isDeleted"] > 0,
+            low_detail_mode=level["isLDM"] > 0,
+            publicity=publicity,
+            demon_difficulty=demon_difficulty,
+            search_flags=search_flag,
+        )
+
+
+async def main() -> int:
+    logger.info("Starting the GMDPS -> RealistikGDPS converter.")
+    ctx = await get_context()
+
+    logger.info("Successfully connected!")
+
+    try:
+        if not await repositories.song.get_count(ctx):
+            logger.info("Converting songs...")
+            await convert_songs(ctx)
+        else:
+            logger.info("Skipping song conversion, songs already exist.")
+
+        if not await repositories.user.get_count(ctx):
+            logger.info("Converting users...")
+            await convert_users(ctx)
+        else:
+            logger.info("Skipping user conversion, users already exist.")
+
+        if not await repositories.level.get_count(ctx):
+            logger.info("Converting levels...")
+            await convert_levels(ctx)
+        else:
+            logger.info("Skipping level conversion, levels already exist.")
+
+        if not await repositories.user_comment.get_count(ctx):
+            logger.info("Converting user comments...")
+            await convert_user_comments(ctx)
+        else:
+            logger.info(
+                "Skipping user comment conversion, user comments already exist.",
+            )
+
+        if not await repositories.level_comment.get_count(ctx):
+            logger.info("Converting level comments...")
+            await convert_level_comments(ctx)
+        else:
+            logger.info(
+                "Skipping level comment conversion, level comments already exist.",
+            )
+    except Exception:
+        logger.error(
+            "Failed to convert data with exception:\n" + traceback.format_exc(),
+        )
+
+    logger.info("Migration complete!")
+    # TODO: Look into a better approach to stop docker
+    # from restarting the container.
+    while True:
+        await asyncio.sleep(100)
```

Comment:
get a list of tasks and use asyncio.gather(*tasks)

Reply:
This is moreso to counter the auto-restart logic of docker-compose if the container exits (so we don't have a chance for migrations to run again).